cmd/scollector/collectors/aws.go

```diff
@@ -0,0 +1,302 @@
+
+func AWS(accessKey, secretKey, region string) {
+	collectors = append(collectors, &IntervalCollector{
+		F: func() (opentsdb.MultiDataPoint, error) {
+			return c_aws(accessKey, secretKey, region)
+		},
+		Interval: 60 * time.Second,
+		name:     fmt.Sprintf("aws-%s", region),
+	})
+}
+
+func c_aws(accessKey, secretKey, region string) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	creds := aws.Creds(accessKey, secretKey, "")
+	if creds == nil {
+		slog.Info("Unable to make creds")
+	}
+	ecc := ec2.New(creds, region, nil)
+	if ecc == nil {
+		slog.Info("Unable to login to EC2")
+	}
+	elb := elb.New(creds, region, nil)
+	if elb == nil {
+		slog.Info("Unable to login to ELB")
+	}
+	cw := cloudwatch.New(creds, region, nil)
+	if cw == nil {
+		slog.Info("Unable to login to CloudWatch")
+	}
+	instances, err := AWSGetInstances(*ecc)
+	if err != nil {
+		slog.Info("Unable to get AWS Elastic Compute Cloud Instances.")
+	}
+	loadbalancers, err := AWSGetLoadBalancers(*elb)
+	if err != nil {
+		slog.Info("Unable to get AWS Elastic Load Balancers.")
+	}
+	for _, loadbalancer := range loadbalancers {
+		AWSGetELBLatency(*cw, &md, loadbalancer)
+		AWSGetELBHostCounts(*cw, &md, loadbalancer)
+	}
+	for _, instance := range instances {
+		AWSGetCPU(*cw, &md, instance)
+		AWSGetNetwork(*cw, &md, instance)
+		AWSGetDiskBytes(*cw, &md, instance)
+		AWSGetDiskOps(*cw, &md, instance)
+		AWSGetStatusChecks(*cw, &md, instance)
+	}
+	return md, nil
+}
+
+func AWSGetInstances(ecc ec2.EC2) ([]ec2.Instance, error) {
+	instancelist := []ec2.Instance{}
+	resp, err := ecc.DescribeInstances(nil)
+	if err != nil {
+		var Error error
+		Error = fmt.Errorf("Unable to describe EC2 Instances")
+		return nil, Error
+	}
+	for _, reservation := range resp.Reservations {
+		for _, instance := range reservation.Instances {
+			instancelist = append(instancelist, instance)
+		}
+	}
+	return instancelist, nil
+}
+
+func AWSGetLoadBalancers(lb elb.ELB) ([]elb.LoadBalancerDescription, error) {
+	lblist := []elb.LoadBalancerDescription{}
+	resp, err := lb.DescribeLoadBalancers(nil)
+	if err != nil {
+		var Error error
+		Error = fmt.Errorf("Unable to describe ELB Balancers")
+		return nil, Error
+	}
+	for _, loadbalancer := range resp.LoadBalancerDescriptions {
+		lblist = append(lblist, loadbalancer)
+	}
+	return lblist, nil
+}
+
+func AWSGetCPU(cw cloudwatch.CloudWatch, md *opentsdb.MultiDataPoint, instance ec2.Instance) {
+	search := cloudwatch.GetMetricStatisticsInput{
+		StartTime:  time.Now().UTC().Add(time.Second * -600),
+		EndTime:    time.Now().UTC(),
+		MetricName: aws.String("CPUUtilization"),
+		Period:     aws.Integer(60),
+		Statistics: []string{"Average"},
+		Namespace:  aws.String("AWS/EC2"),
+		Unit:       aws.String("Percent"),
+		Dimensions: []cloudwatch.Dimension{{Name: aws.String("InstanceId"), Value: instance.InstanceID}},
+	}
+	resp, err := cw.GetMetricStatistics(&search)
+	if err != nil {
+		slog.Info("Error getting Metric Statistics: ", err)
```

Comment:
Probably need a return here, instead of a slog.Info. The function can either return an error (if it expected to succeed), or return some empty data (if failure is acceptable and should just be silently ignored).